tests/ci/build_download_helper.py

```diff
@@ -89,6 +89,7 @@ def set_auth_header():
                 set_auth_header()
                 need_retry = True
                 break
+            raise exc
```

Comment:
Accidentally added in https://github.com/ClickHouse/ClickHouse/commit/1fbfcbc3d3f9d7bd047d8ae2da8b658c28249e15#r129706939

It neglects `for _ in range(retries):` from line 75, so it shouldn't be here

Reply:
```
In [3]: for i in range(5):
   ...:     print(i)
   ...: else:
   ...:     print('finished')
   ...:
0
1
2
3
4
finished

In [4]: for i in range(5):
   ...:     print(i)
   ...:     if i > 3:
   ...:         break
   ...: else:
   ...:     print('finished')
   ...:
0
1
2
3
4
```

It looks like we can use `else` here